src/creddit/client.clj

```diff
@@ -218,3 +242,29 @@
   [credentials names]
   (-> (http-get credentials (str "https://www.reddit.com/by_id/" (string/join "," names) "/.json"))
       (parse-response)))
+
+; other functions use credentials directly, unsure why mine needs to get them from the client with get-in...
+(defn submit
+  [credentials subreddit kind title content]
+  (-> (client/post "https://oauth.reddit.com/api/submit"
+                   (cond (= kind "link")
```

Comment:
The value of the two cond expressions are basically identical except for the value of `:kind` and the keys `:url` or `:text`.

`kind` is being passed in so we can say `{:kind kind}` as for `:url` and `:text` I think we could use a `let` and then either a `cond` or `merge` to add they correct key? What do you think?

Reply:
I updated it.